WordPress/gutenberg
# [RNMobile] Ensure dictating text doesn't cause cursor to be offset on iOS

Fixes https://github.com/WordPress/gutenberg/issues/51227

## Related PRs

* `gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5868
* [Testing purposes only] `WordPress-iOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/20881

## What?

The changes in this PR ensure that the cursor's position is always correct during dictation on iOS.

## Why?

While dictating text in the editor on iOS, there are currently some occasions where the cursor's position is either off-by-one, or when it briefly appears out of place on a new line. It'd be preferable for the cursor to always remain in the correct position.

## How?

This PR introduces a `newPositionOffset` variable that captures whether the cursor's position is off-by-one. If it is, one is subtracted to ensure the cursor is always at the correct position.

## Testing Instructions

> **Note**
> An installable build is available at https://github.com/wordpress-mobile/WordPress-iOS/pull/20881 for easier testing, if preferred

* In the iOS app, navigate to _My Site_ → _Posts_ and create a new post.
* Tap on "Start writing..." to focus the default paragraph block.
* Activate the iOS dictation feature and dictate "Mary had a little lamb its fleece was white as snow".
* Place the cursor after "little" then deactivate iOS dictation.
* Verify that cursor remains in place and doesn't move by one space.
* Activate the iOS dictation feature again and dictate "Old MacDonald had a Farm"
* Verify that the cursor's placement remains in place directly after "Farm".
* Continue dictating and verify that the cursor's placement is always correct.
